hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -743,21 +791,124 @@ func (r *HostedClusterReconciler) reconcileIgnitionServer(ctx context.Context, h
 		}
 		ignitionServerService.Spec.Type = corev1.ServiceTypeClusterIP
 		return nil
-	})
+	}); err != nil {
+		return fmt.Errorf("failed to reconcile ignition service: %w", err)
+	}
 
 	// Reconcile route
 	ignitionServerRoute := ignitionserver.Route(controlPlaneNamespace.Name)
-	_, err = controllerutil.CreateOrUpdate(ctx, r.Client, ignitionServerRoute, func() error {
+	if _, err := controllerutil.CreateOrUpdate(ctx, r.Client, ignitionServerRoute, func() error {
+		ignitionServerRoute.Annotations = map[string]string{
+			hostedClusterAnnotation: ctrlclient.ObjectKeyFromObject(hcluster).String(),
+		}
+		ignitionServerRoute.Spec.TLS = &routev1.TLSConfig{
+			Termination: routev1.TLSTerminationPassthrough,
+		}
 		ignitionServerRoute.Spec.To = routev1.RouteTargetReference{
 			Kind: "Service",
 			Name: ignitionserver.ResourceName,
 		}
 		return nil
-	})
+	}); err != nil {
+		return fmt.Errorf("failed to reconcile ignition route: %w", err)
+	}
+
+	// The route must be admitted and assigned a host before we can generate certs
+	if len(ignitionServerRoute.Status.Ingress) == 0 || len(ignitionServerRoute.Status.Ingress[0].Host) == 0 {
+		r.Log.Info("ignition server reconciliation waiting for ignition server route to be assigned a host value")
+		return nil
+	}
+
+	// Reconcile a root CA for ignition serving certificates. We only create this
+	// and don't update it for now.
+	caCertSecret := ignitionserver.IgnitionCACertSecret(controlPlaneNamespace.Name)
+	if result, err := controllerutil.CreateOrUpdate(ctx, r.Client, caCertSecret, func() error {
+		if caCertSecret.CreationTimestamp.IsZero() {
+			cfg := &certs.CertCfg{
+				Subject:   pkix.Name{CommonName: "ignition-root-ca", OrganizationalUnit: []string{"openshift"}},
+				KeyUsages: x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature | x509.KeyUsageCertSign,
+				Validity:  certs.ValidityTenYears,
+				IsCA:      true,
+			}
+			key, crt, err := certs.GenerateSelfSignedCertificate(cfg)
+			if err != nil {
+				return fmt.Errorf("failed to generate CA: %w", err)
+			}
+			caCertSecret.Type = corev1.SecretTypeTLS
+			caCertSecret.Data = map[string][]byte{
+				corev1.TLSCertKey:       certs.CertToPem(crt),
+				corev1.TLSPrivateKeyKey: certs.PrivateKeyToPem(key),
+			}
+		}
+		return nil
+	}); err != nil {
+		return fmt.Errorf("failed to reconcile ignition ca cert: %w", err)
+	} else {
```

Comment:
nit: This else block wouldn't be actually needed if we don't inline `if result, err :=` right? Just wanted to point it to confirm, I'm good with both styles.

Reply:
Without inlining the if block, `result` would still only be safe to use if `err != nil` so I'm not sure the else is avoidable